fixed the slice and reverse op to support tensorarray on x86

wbn03 opened the pull request:
当前修改未完成，麻烦先别合入

Commit:
fixed the slice op to support tensorarray on x86

Commit:
fixed the slice op to support tensorarray on x86

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle-Lite?pullRequest=9089) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle-Lite?pullRequest=9089) before we can accept your contribution.<br/>**6** out of **7** committers have signed the CLA.<br/><br/>:white_check_mark: chenjiaoAngel<br/>:white_check_mark: zhoutianzi666<br/>:white_check_mark: yingshengBD<br/>:white_check_mark: csy0225<br/>:white_check_mark: shentanyue<br/>:white_check_mark: affectionlu<br/>:x: wangbingnan03<br/><hr/>**wangbingnan03** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle-Lite?pullRequest=9089) it.</sub>

chenjiaoAngel committed:
[arm] add v9 gemm implementation (#9083)

csy0225 committed:
Autoscan support xpu demo (#9081)

* autoscan support xpu test=huawei_ascend_npu,test=intel_openvino,test=cambricon_mlu

* xpu autoscan demo

yingshengBD committed:
[Doc]TIM-VX & amlogicDDK, document fix (#9050)

* [Doc]TIM-VX & amlogicDDK, document fix
test=document_fix

* [Doc]TIM-VX & amlogicDDK, document fix
test=document_fix

* [Doc]TIM-VX & amlogicDDK, document fix
test=document_fix

* [Doc]TIM-VX & amlogicDDK, document fix
test=document_fix

* [Doc]TIM-VX & amlogicDDK, document fix
test=document_fix

zhoutianzi666 committed:
[Op] fix split bug when sections having -1 (#8934)

shentanyue committed:
[Op][Pass] Support Swin-Transformer model (#9065)

affectionlu committed:
[NNAdapter][IntelOpenVINO] Add support for model cache to reduce memory consumption and first inference time, (#9088)

test=intel_openvino

affectionlu committed:
[NNAdapter][IntelOpenvino] Add mat_mul, dropout op (#9082)

test=intel_openvino

shentanyue reviewed (dismissed):
LGTM

shentanyue commented:
签署下CLA，以及关注下CI，包括Lite-CodeStyle-Precommit，最好在我们lite的容器内提交代码，clang-format版本需要3.8
详见：https://paddle-lite.readthedocs.io/zh/develop/develop_guides/add_hardware.html#review-ci
